### PR TITLE
Articles: Clojures

### PR DESCRIPTION
**Pull request by llamatarianism**
Adds 5 articles about Clojure (a functional dialect of Lisp) to the Wiki:
* Clojure-Summary
   An article about Clojure, and why someone might want to use it.
* Clojure-Basics
   An article about the basics of Clojure: defining variables and functions.
* Clojure-Conditionals
   An article about logical operators and if/else in Clojure.
* Clojure-Let-Bindings
   An article about using let to define local variables.
* Clojure-Loop-Recur
   An article about using recursion (and in particular, the loop/recur syntax).

### CODE REVIEW

**Comment by Rafase282** [comment]
The filenames are great. I would suggest you create a `Cloujure.md` that would be where all related articles would be linked to.

File: Clojure-Basics.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+...Hold on. `(+ x y z)`? That looks funny. Well, Lisps are written using "prefix
+notation", which means that the function always comes first. Since all
+mathematical operators in Lisp (`+ - * /`) are just functions, they also come
+before their arguments (in this case, `x y z`).
+
+You'll notice that our vector has some stuff in it now: `[x y z]`! Whenever
+a function has parameters, you define them in that vector next to the function's
+name.
+
+ ###### Destructuring
```
Comment (Rafase282) [inline Clojure-Basics.md]: This does not parse.

File: Clojure-Basics.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+```clojure
+(defn add [[x y] z] (+ x y z))
+(add [1 2] 3)
+; => 6
+```
+The arguments to that function are a collection ([`x y`]) and a number (`z`).
+We can use destructuring to pull the first and second elements out of the list,
+and call them `x` and `y`.
+
+###### &
```
Comment (Rafase282) [inline Clojure-Basics.md]: I'm not sure about the purpose of this header.

File: Clojure-Conditionals.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
+# Clojure - Conditionals
+
+You're not going to get anywhere in a language if all you can do is define
+functions that print things and do simple arithmetic. Conditionals and logic
+are a fundamental part of making code that does interesting, useful things. Try
+and imagine a world without logic in programs: you wouldn't even be able to simple
+things, like checking if two numbers are equal!
+
+### Logical Operators
```
Comment (Rafase282) [inline Clojure-Conditionals.md]: Try to keep the heading in ascending order if possible.

After `#` all other top level headings should be `##` and bellow those, the top ones should be `###` and so on.

**Comment by alayek** [comment]
I can't express my happiness over your pull request. The only thing I find missing, is an introductory articl, in `Clojure.md`.

Thanks @llamatarianism

**Comment by iheartkode** [comment]
I agree with @alayek Nice addition to the wiki :+1:

**Comment by theflametrooper** [comment]
Add Navigation Buttons shown [here](https://github.com/FreeCodeCamp/wiki/issues/816#issuecomment-216679394) that shows previous, next and home.

File: Clojure-Basics.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+familiar with this concept. Rather than telling our function to return something,
+it evaluates all the code inside its body, and returns the result. Our `add`
+function, for example, evaluates `(+ x y z)`, and then returns that result.
+
+The reason why our functions that use `println` output `nil` is because `println`
+evaluates to `nil`. (`nil` is like `null` or `None` -- it represents nothing.)
+
+| [:point_left: Previous](https://github.com/FreeCodeCamp/wiki/blob/master/Clojure-Summary.md) | [:book: Home :book:](https://github.com/FreeCodeCamp/wiki/blob/master/Clojure.md) | [Next :point_right:](https://github.com/FreeCodeCamp/wiki/blob/master/Clojure-Conditionals.md)|
+|:---|:---:|----:|
+| [Summary](https://github.com/FreeCodeCamp/wiki/blob/master/Clojure-Summary.md) | [Table of Contents](https://github.com/FreeCodeCamp/wiki/blob/master/Clojure.md) | [Conditionals](https://github.com/FreeCodeCamp/wiki/blob/master/Clojure-Conditionals.md)|
```
Comment (theflametrooper) [inline Clojure-Basics.md]: ~~`https://github.com/FreeCodeCamp/wiki/blob/master/Clojure.md`~~ links can just be the name of file without extension (`.md`).

For example:
```markdown
| [:point_left: Previous](Clojure-Summary) | [:book: Home :book:](Clojure) | [Next :point_right:](Clojure-Conditionals)|
|:---|:---:|----:|
| [Summary](Clojure-Summary) | [Table of Contents](Clojure) | [Conditionals](Clojure-Conditionals)|
```

File: Clojure.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+[Clojure](https://clojure.org/) is a dialect of Lisp with a focus on functional programming.
+It was released in 2007 by Rich Hickey, and has since gained a lot of popularity
+among web developers.
+
+To learn about some of the language's core functions, you can consult the
+[documentation.](https://clojuredocs.org/)
+
+You could also consult any of these articles:
+* [Summary](Clojure-Summary.md) - Why use Clojure?
```
Comment (Rafase282) [inline Clojure.md]: You don't need to add `.md` for relative links.

**Comment by theflametrooper** [comment]
## LGTM :+1:

Just waiting for @Rafase282 
Great article @llamatarianism :+1:

**Comment by Rafase282** [comment]
## Thank you for the contribution.

**Pull request merged**